onnxruntime/python/tools/transformers/models/bart/export.py

```diff
@@ -0,0 +1,96 @@
+import argparse
+import logging
+import os
+import sys
+
+import torch
+from utils import (
+    chain_enc_dec_with_beamsearch,
+    export_summarization_edinit,
+    export_summarization_enc_dec_past,
+    onnx_inference,
+)
+
+# GLOBAL ENVS
+logging.basicConfig(
+    format="%(asctime)s | %(levelname)s | %(name)s |  [%(filename)s:%(lineno)d] %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+    level=os.environ.get("LOGLEVEL", "INFO").upper(),
+    stream=sys.stdout,
+)
+logger = logging.getLogger("generate")
+
+
+def print_args(args):
+    for arg in vars(args):
+        logger.info(f"{arg}: {getattr(args, arg)}")
+
+
+def user_command():
+
+    parent_parser = argparse.ArgumentParser(add_help=False)
+    parent_parser.add_argument("--max_length", type=int, default=20, help="default to 20")
+    parent_parser.add_argument("--min_length", type=int, default=0, help="default to 0")
+    parent_parser.add_argument("-o", "--output", type=str, default="onnx_models", help="default name is onnx_models.")
+    parent_parser.add_argument("-i", "--input_text", type=str, default=None, help="input text")
+    parent_parser.add_argument("-s", "--spm_path", type=str, default=None, help="tokenizer model from sentencepice")
+    parent_parser.add_argument("-v", "--vocab_path", type=str, help="vocab dictionary")
+    parent_parser.add_argument("-nb", "--num_beams", type=int, default=5, help="default to 5")
+    parent_parser.add_argument("-rp", "--repetition_penalty", type=float, default=1.0, help="default to 1.0")
+    parent_parser.add_argument("-ns", "--no_repeat_ngram_size", type=int, default=3, help="default to 3")
+    parent_parser.add_argument("-es", "--early_stopping", type=bool, default=False, help="default to False")
+    parent_parser.add_argument("-op", "--opset_version", type=int, default=14, help="default to 14")
+    parent_parser.add_argument("--cuda", action="store_true", help="use CUDA")
```

Comment:
what happens when the system has several cuda libraries installed? Will it use `CUDA_HOME`? Should this flag take path to cuda instead of a bool? Empty means no cuda and any string is used as CUDA_HOME and set the system with it

Reply:
CUDA is actually not supported in this case. It was meant for PyTorch usage only, but I took GPU option away anyway.